backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/gson/impl/JsonObjectAccessorImpl.java

```diff
@@ -37,4 +41,17 @@ public UnknownTypeJsonAccessor property(String propertyName) {
 		return new ObjectPropertyJsonAccessor( this, propertyName );
 	}
 
+	@Override
+	public UnknownTypeJsonAccessor path(String dotSeparatedPath) {
+		String[] components = DOT_REGEX.split( dotSeparatedPath );
```

Comment:
Wondering if we should add a `Contracts#assertNotNullNorEmpty()` call for this method and the method above.

Reply:
I don't think it can happen in this specific case, but it cannot hurt to check. Added a commit, will merge as soon as CI is green.